starter-code/scripts/article.js

```diff
@@ -50,13 +50,22 @@ Article.fetchAll = function() {
     // When rawData is already in localStorage,
     // we can load it with the .loadAll function above,
     // and then render the index page (using the proper method on the articleView object).
-    Article.loadAll(); //TODO: What do we pass in to loadAll()?
-    //TODO: What method do we call to render the index page?
+    Article.loadAll(JSON.parse(localStorage.rawData)); //DONE: What do we pass in to loadAll()?
+    //DONE: What method do we call to render the index page?
+    articleView.initIndexPage();
   } else {
-    // TODO: When we don't already have the rawData,
+    // DONE: When we don't already have the rawData,
     // we need to retrieve the JSON file from the server with AJAX (which jQuery method is best for this?),
     // cache it in localStorage so we can skip the server call next time,
     // then load all the data into Article.all with the .loadAll function above,
     // and then render the index page.
+    $.getJSON('/data/hackerIpsum.json')
+    .then(function(data) {
+      localStorage.rawData = JSON.stringify(data);
+    }, function(err) {
+      console.error('Error: ' + err);
+    });
+    Article.loadAll(JSON.parse(localStorage.rawData));
```

Comment:
This currently isn't working when you don't already have the data loaded. You are properly loading the data to local storage but because your load all and init Index Page are outside of the then block you are actually calling them before you get your response from the server. You should move those 2 lines inside the then block. This will also allow you to use something more semantic than pulling down local storage right after you set it.